Sources/NIO/GetaddrinfoResolver.swift

```diff
@@ -79,7 +79,11 @@ internal class GetaddrinfoResolver: Resolver {
     private func resolve(host: String, port: Int) {
         var info: UnsafeMutablePointer<addrinfo>?
 
-        guard getaddrinfo(host, String(port), nil, &info) == 0 else {
+        // We only use this resolver for TCP SocketChannel so set some hints.
+        var hint = addrinfo()
+        hint.ai_socktype = Posix.SOCK_STREAM
+        hint.ai_protocol = Posix.IPPROTO_TCP
+        guard getaddrinfo(host, String(port), &hint, &info) == 0 else {
```

Comment:
What about just allow to pass this via the init?

Reply:
Sure, that'd be fine too, but again, it's gotta be on the protocol itself (so that other hypothetical resolvers can handle this in the future).

Reply:
@Lukasa I don't see why we would need to have a protocol for this (at least for now). We can add it later once we start to add other resolvers IMHO.

Reply:
I mean, the only reason to put it on the protocol is because it’s presumably just as relevant for other resolvers as for this one. But I’m ok deferring it I guess.

Reply:
Yeah I think I would defer it for now and care about once we start with the async resolver